### PR TITLE
Detect import path from Godeps.json

### PR DESCRIPTION
**Pull request by twaugh**
Signed-off-by: Tim Waugh <twaugh@redhat.com>

### CODE REVIEW

File: backvendor/gosource.go [hunk 1]
```diff
@@ -93,6 +102,38 @@ func NewGoSource(pth string, excludeGlobs ...string) (*GoSource, error) {
 	return src, nil
 }
 
+// readGodepsConf parses Godeps/Godeps.json to extract the package
+// name.
+func readGodepsConf(src *GoSource) error {
```
Comment (mprahl) [inline backvendor/gosource.go]: I'm not sure if this is usual in Go, but when I read a function called `readGodepsConf`, I don't assume it's going to change the value of the struct I passed in. Perhaps a better name for this function is `loadGodepsConf`.

File: backvendor/gosource.go [hunk 2]
```diff
@@ -93,6 +102,38 @@ func NewGoSource(pth string, excludeGlobs ...string) (*GoSource, error) {
...
+		ImportPath string
+	}
+	conf := filepath.Join(src.Path, "Godeps", "Godeps.json")
+	if _, skip := src.excludes[conf]; skip {
+		return nil
+	}
+	f, err := os.Open(conf)
+	if err != nil {
+		if os.IsNotExist(err) {
+			err = nil
```
Comment (mprahl) [inline backvendor/gosource.go]: Optional: For readability, I think it's better to just return `nil` here rather than overwrite a variable's value.

**Review by mprahl — APPROVED** [review]

File: backvendor/gosource.go (hunk 1 above)
Comment (twaugh) [reply backvendor/gosource.go]: Good idea, renamed.

File: backvendor/gosource.go (hunk 2 above)
Comment (twaugh) [reply backvendor/gosource.go]: Agreed, fixed.

**Review by mprahl — APPROVED** [review]
:+1:

**Pull request merged**